### DISCUSSION TITLE
mutateFormDataBeforeFill Method for Create

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#2511] [Ideas]
It might be helpful to have a `mutateFormDataBeforeFill()` method when creating a new record. My use case: I have a system with exhibitors and registrants and I want to have the ability to create a new registrant for an exhibitor, where some of the new registrant data is pre-filled from the exhibitor record.

**Comment by ryangjchandler** [comment]
@intrepidws Couldn't you use a model event/hook for this instead?

**Reply by intrepidws** [reply]
@ryangjchandler I'd like the form itself to show the values as a default, so a model event wouldn't work. But you're right that I should be able to use the `afterFill` hook. Thanks.

**Reply by danharrin** [reply]
`mutateFormDataBeforeFill()` wouldnt be appropriate here, as filling the form would then just clear all the values again (since form component defaults are `null`)

**Comment by rajatpatelz** [comment]
You can use do this by first registering a Relationship Manager and then using Closure to fill default value of form field like this
```
...
Forms\Components\TextInput::make('exhibitor_name')->default( function (Livewire $livewire) {
    return $livewire->ownerRecord->name;
} );
...
```
